### PR TITLE
Hacktoberfest Filter for status

### PR DESCRIPTION
**Pull request by dubrsl**
- Added filter by target branch or name for status notification
- Removed SkipNotificationsTrait because new PipelineStatusStrategy has NONE option and duplicate it
- Used getStatusName in onEnterWaiting function
- Beautified GitLabPipelineStatusNotifier.java

### CODE REVIEW

**Comment by dubrsl** [comment]
IMHO you can close #96 and #97 because this PR contain code from those PRs

File: pom.xml [hunk 1]
```diff
@@ -175,6 +175,24 @@
       <groupId>org.jenkins-ci.plugins.workflow</groupId>
       <artifactId>workflow-job</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-job</artifactId>
+      <version>2.39</version>
+      <scope>compile</scope>
```
Comment (oleg-nenashev) [inline pom.xml:182]: As commented somewhere before, this is a bad idea to use compile scope

Comment (dubrsl) [reply pom.xml]: Fixed

**Comment by sichapman** [comment]
Is there any possibility of this MR moving forward? Speaking somewhat selfishly, the `markUnstableAsSuccess` portion of this MR would be incredibly useful for us!

**Comment by usinelogicielle** [comment]
This PR seems a great things to adapt status between Jenkins and Gitlab

**Comment by jetersen** [comment]
I am unsure why you created another PR.

Once you open a PR you can add more commits by pushing to the same branch.

Either close this or fixup this PR and close the other PR.

Also the PRs has conflict, would you be able to resolve them?

**Pull request closed without merging**